src/common/ownsql.cpp

```diff
@@ -490,18 +490,28 @@ void SqlQuery::reset_and_clear_bindings()
     }
 }
 
-bool SqlQuery::initOrReset(const QByteArray &sql, OCC::SqlDatabase &db)
+PreparedSqlQueryRAII::PreparedSqlQueryRAII(SqlQuery *query)
+    : _query(query)
 {
-    ENFORCE(!_sqldb || &db == _sqldb);
-    _sqldb = &db;
-    _db = db.sqliteDb();
-    if (_stmt) {
-        reset_and_clear_bindings();
-        return true;
-    } else {
-        return prepare(sql) == 0;
+    Q_ASSERT(!sqlite3_stmt_busy(_query->_stmt));
+}
+
+PreparedSqlQueryRAII::PreparedSqlQueryRAII(SqlQuery *query, const QByteArray &sql, SqlDatabase &db)
+    : _query(query)
+{
+    Q_ASSERT(!sqlite3_stmt_busy(_query->_stmt));
```

Comment:
@mgallien It looks confusing that we are working with a member `_query` here, then, with parameter `query` again.